libkbfs/test_stallers.go

```diff
@@ -38,6 +39,156 @@ const (
 	StallableMDPutUnmerged           StallableMDOp = "PutUnmerged"
 )
 
+const stallKeyStallEveryThing = ""
+
+type naïveStallInfo struct {
+	onStalled   <-chan struct{}
+	unstall     chan<- struct{}
+	oldBlockOps BlockOps
+	oldMDOps    MDOps
+}
+
+// NaïveStaller is used to stall certain ops in BlockOps or MDOps. Unlike
+// StallBlockOp and StallMDOp which provides a way to precisely control which
+// particular op is stalled by passing in ctx with corresponding stallKey,
+// NaïveStaller simply stalls all instances of specified op.
+type NaïveStaller struct {
+	config Config
+
+	mu             sync.RWMutex
+	blockOpsStalls map[StallableBlockOp]*naïveStallInfo
```

Comment:
My rationale for preferring value types here is that returning pointers to lock-protected structures carry the danger that something might inadvertently modify them outside the lock.

But I don't think it matters too much in this case, so I'm okay with leaving it as pointers.